### DISCUSSION TITLE
Utilizing column values in Javascript Tooltip Generator/Onclick HREF

### DISCUSSION THREAD

**Discussion by ArmadilloDisco** [apache/superset#24105] [Q&A / Help] [answered]
I cannot find any documentation on how to actually reference columns using the 'Extra Data for JS' feature when making a Deck.gl map.

So for instance, as a test I am using the public.sf_population_polygon data in the example data sets to generate a map that when you click in one of the zipcode areas, it will take you to another dashboard with that zipcode applied as a filter.

I have added the zipcode column in the 'Extra Data for JS' options.
So in the Javascript Onclick HREF, I followed one example I found via google that showed to do something like this:
`function(){
  let base_string = "/superset/dashboard/12/?zipcode="
  let filter_string = object.object[0].extraProps.zipcode
  filter_string = filter_string.toString()
  filter_string = base_string.concat(filter_string)
  return filter_string
}`

With the zipcode of the clicked polygon being referenced by 'object.object[0].extraProps.zipcode'

However just returns "/superset/dashboard/12/?zipcode=" 

How do I reference the columns in the 'Extra Data for JS' feature?

**Comment by ArmadilloDisco** [comment, marked as answer]
After more trial and error and combining ideas from other answers, found this that worked:
`d => "/superset/dashboard/12/?zipcode=" + d.object.extraProps.zipcode`

**Comment by Stefan781** [comment]
I am also struggling with the lack of documentation for the JS for Deck.gl visualizations. It is more of a reverse engineering than it is developing when using it. Would be useful if there is some clarification on how it works with a few examples in the documentation.

**Comment by neers-0901** [comment]
I need to open two URLs from Deck.gl. I am able to open with windows.open. But the thing is I need to pass dynamic paramters based on clicked point. Below is my code. Any help would be greatly appreciated.

d => "https://latsuperset1.asia.abi.dyn.nesc.nokia.net/superset/dashboard/203/?sitename=" + d.object.extraProps.sitename + "&circle_name=" + d.object.extraProps.circle_name
window.open("https://www.google.com/"); 
e => "https://latsuperset1.asia.abi.dyn.nesc.nokia.net/superset/dashboard/199/,sitename=" + e.object.extraProps.sitename
